app/components/primer/timeline_item_component.rb

```diff
@@ -1,22 +1,49 @@
 # frozen_string_literal: true
 
+require_relative "timeline_item/badge_component"
+
 module Primer
   # Use `TimelineItem` to display items on a vertical timeline, connected by badge elements.
   class TimelineItemComponent < Primer::Component
-    include ViewComponent::Slotable
+    include ViewComponent::SlotableV2
+
+    # Avatar to be rendered to the left of the Badge.
+    #
+    # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::AvatarComponent) %>.
+    renders_one :avatar, lambda { |src:, size: 40, square: true, **system_arguments|
+      system_arguments[:classes] = class_names(
+        "TimelineItem-avatar",
+        system_arguments[:classes]
+      )
+
+      Primer::AvatarComponent.new(src: src, size: size, square: square, **system_arguments)
+    }
 
-    with_slot :avatar, class_name: "Avatar"
-    with_slot :badge, class_name: "Badge"
-    with_slot :body, class_name: "Body"
+    # Badge that will be connected to other TimelineItems.
+    #
+    # @param icon [String] Name of [Octicon](https://primer.style/octicons/) to use.
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :badge, Primer::TimelineItem::BadgeComponent
```

Comment:
For my own learning, why did you choose to make badge its own class instead of using a lambda like with the other slots?

Reply:
because it was more complex. the `badge` slot also defines its own content (an octicon) so I thought it was better to keep it as a separate component.

I don't think we can declare content for the component in a lambda, correct me if I'm wrong @BlakeWilliams

Reply:
If `BadgeComponent` is only ever used in this context, IMO we should try to embed it inside this component. I'd also be curious to hear @BlakeWilliams' thoughts 😄

Reply:
I moved `BadgeComponent` to live inside `TimelineItemComponent` :)